WordPress/Classes/ViewRelated/Settings/AboutViewController.swift

```diff
@@ -0,0 +1,230 @@
+            return CGFloat.min
+        }
+        
+        return footerView!.bounds.height
+    }
+
+    public override func tableView(tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        if section != (rowTitles.count - 1) {
+            return nil
+        }
+        
+        return footerView
+    }
+    
+    public override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        tableView.deselectSelectedRowWithAnimation(true)
+        
+        if let handler = rowHandlers[indexPath.section][indexPath.row] {
+            handler()
+        }
+    }
+    
+    
+    
+    // MARK: - Private Helpers
+    private func displayWebView(url: String) {
+        let webViewController = WPWebViewController(URL: NSURL(string: url)!)
+        if presentingViewController != nil {
+            navigationController?.pushViewController(webViewController, animated: true)
+        } else {
+            let navController = UINavigationController(rootViewController: webViewController)
+            presentViewController(navController, animated: true, completion: nil)
+        }
+    }
+
+    private func displayRatingPrompt() {
+        // Note: 
+        // Let's follow the same procedure executed as in NotificationsViewController, so that if the user
+        // manually decides to rate the app, we don't render the prompt!
+        //
+        WPAnalytics.track(WPAnalyticsStat.AppReviewsRatedApp)
+        AppRatingUtility.ratedCurrentVersion()
+        ABXAppStore.openAppStoreForApp(WPiTunesAppId)
+    }
+    
+    private func displayTwitterComposer() {
+        if isTwitterUnavailable() {
+            return
+        }
+
+        let twitterURL = NSURL(string: WPTwitterWordPressMobileURL)!
+        UIApplication.sharedApplication().openURL(twitterURL)
+
+    }
+    
+    
+    // MARK: - Twitter Helpers
+    private func isTwitterUnavailable() -> Bool {
+        let emptyTwitterURL = NSURL(string: WPTwitterWordPressMobileURL)!
+        return UIApplication.sharedApplication().canOpenURL(emptyTwitterURL) == false
+    }
+    
+    private func filterDisabledRows<T>(array: [[T]]) -> [[T]] {
+        var filtered = array
+        
+        if isTwitterUnavailable() {
+            var section = array[twitterIndexPath.section] as [T]
+            section.removeAtIndex(twitterIndexPath.row)
+            filtered[twitterIndexPath.section] = section
+        }
+        
+        return filtered
+    }
+    
+    
+
+    // MARK: - Private Constants
+    private let reuseIdentifier         = "reuseIdentifierValue1"
+    private let twitterIndexPath        = NSIndexPath(forRow: 0, inSection: 1)
+    private let iconBottomPadding       = CGFloat(30)
+    
+    // MARK: - Private Aliases
+    typealias RowHandler = (Void -> Void)
+    
+    // MARK: - Private Properties
+    private var footerView : WPTableViewSectionFooterView!
+    
+    private var rowTitles : [[String]] {
```

Comment:
I'm not sure having three methods to describe the row data, each building its own array, is the best approach. Would it make more sense to declare a type with the relevant properties and have a single array?  Seems like this would be more straight forward to update in the future.